File: 0682/sesión25 - Reportes/Front-End/SoftInv/SoftInvBusiness/AlmacenBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using SoftInvBusiness.almacenesWS;

namespace SoftInv.BO
{
    public class AlmacenBO
    {
        private AlmacenesClient almacenesClienteSOAP;

        public AlmacenBO()
        {
            this.almacenesClienteSOAP = new AlmacenesClient();
        }

        public int Insertar(string nombre, bool almacenCentral)
        {
            return this.almacenesClienteSOAP.insertar(nombre, almacenCentral);
        }

        public int Eliminar(int almacenId)
        {
            return this.almacenesClienteSOAP.eliminar(almacenId);
        }

        public int Modificar(int almacenId, string nombre, bool almacenCentral)
        {
            return this.almacenesClienteSOAP.modificar(almacenId, nombre, almacenCentral);
        }

        public almacenesDTO ObtenerPorId(int almacenId)
        {
            return this.almacenesClienteSOAP.obtenerPorId(almacenId);
        }

        public BindingList<almacenesDTO> ListarTodos()
        {
            almacenesDTO[] arreglo = this.almacenesClienteSOAP.listarTodos();
            return new BindingList<almacenesDTO>(arreglo);
        }

        public byte[] reporteProductos()
        {
            return this.almacenesClienteSOAP.reporteProductos();
        }

        public byte[] reporteTipoDeProductos(int pTipoProductoId)
        {
            return this.almacenesClienteSOAP.reporteTipoDeProductos(pTipoProductoId);
        }

        public void abrirReporte(HttpResponse response, string nombreReporte, byte[] reporte)
        {
            response.Clear();
            response.ContentType = "application/pdf";
            response.AddHeader("Content-Disposition", "inline;filename=" + nombreReporte + ".pdf");
            response.BinaryWrite(reporte);
            response.End();
        }
    }
}

File: 0682/sesión 19 - SOAP (2)/Back-End/SoftInvBusiness/src/main/java/pe/edu/pucp/softinv/bo/AlmacenBO.java
package pe.edu.pucp.softinv.bo;

import java.util.ArrayList;
import pe.edu.pucp.softinv.dao.AlmacenDAO;
import pe.edu.pucp.softinv.daoImp.AlmacenDAOImpl;
import pe.edu.pucp.softinv.model.AlmacenesDTO;

public class AlmacenBO {

    private AlmacenDAO almacenDAO;
    
    public AlmacenBO(){
        this.almacenDAO = new AlmacenDAOImpl();
    }
    
    public Integer insertar(String nombre, Boolean almacen_central){
        AlmacenesDTO almacenesDTO = new AlmacenesDTO();
        almacenesDTO.setNombre(nombre);
        almacenesDTO.setAlmacen_central(almacen_central);
        return this.almacenDAO.insertar(almacenesDTO);
    }
    
    public AlmacenesDTO obtenerPorId(Integer almacenId){
        AlmacenesDTO almacenesDTO = new AlmacenesDTO();
        almacenesDTO.setAlmacenId(almacenId);
        return this.almacenDAO.obtenerPorId(almacenId);
    }
    
    public ArrayList<AlmacenesDTO> listarTodos(){
        return this.almacenDAO.listarTodos();
    }
    
    public Integer modificar(Integer almacenId, String nombre, Boolean almacen_central){
        AlmacenesDTO almacenesDTO = new AlmacenesDTO();
        almacenesDTO.setAlmacenId(almacenId);
        almacenesDTO.setNombre(nombre);
        almacenesDTO.setAlmacen_central(almacen_central);
        return this.almacenDAO.modificar(almacenesDTO);
    }
    
    public Integer eliminar(Integer almacenId){
        AlmacenesDTO almacenesDTO = new AlmacenesDTO();
        almacenesDTO.setAlmacenId(almacenId);
        return this.almacenDAO.eliminar(almacenesDTO);
    }
}
